src/legacy/server/logging/log_format_string.js

```diff
@@ -70,12 +70,23 @@ export default class KbnLoggerStringFormat extends LogFormat {
     const time = color('time')(this.extractAndFormatTimestamp(data, 'HH:mm:ss.SSS'));
     const msg = data.error ? color('error')(data.error.stack) : color('message')(data.message);
 
-    const tags = _(data.tags)
-      .sortBy(function (tag) {
-        if (color(tag) === _.identity) return `2${tag}`;
-        if (_.includes(statuses, tag)) return `0${tag}`;
-        return `1${tag}`;
-      })
+    const isColor = tag => color(tag) === _.identity;
+    const isStatus = tag => _.includes(statuses, tag);
+    const isContext = tag => tag.__kibanaContext__ === true;
+
+    const contextTags = data.tags.filter(isContext).map(tag => tag.value);
+    const nonContextTags = data.tags.filter(tag => !isContext(tag));
+
+    const coloredTags = nonContextTags.filter(isColor).sort();
+    const statusTags = nonContextTags.filter(isStatus).sort();
+    const otherTags = nonContextTags.filter(tag => !isColor(tag) && !isStatus(tag) && !isContext(tag)).sort();
+
+    const tags = [
+      ...coloredTags,
+      ...statusTags,
+      ...contextTags,
+      ...otherTags,
+    ]
```

Comment:
Follow to previous comment, but imho this is not the responsibility of a log format. What would be the amount of work to properly types the tags instead of having a flat list ?

Thinking of something like
```ts
type LogEntryTags = {
    status: string;
    context: string[];
    // probably other things, don't know everything we did put in there
}
```

That way, the formatter could apply a actual (even if still hardcoded)  format instead of having some identification and sorting logic in there?

Reply:
👍 to evaluate how much work requires introducing a separate log record format for NP. Although I'd avoid putting this under `tag` property. This part in LP isn't covered by TS.
https://github.com/restrry/kibana/blob/b9a43a1788486c0d064cbe7aaff1fc7e987f7dfa/src/core/server/legacy/logging/legacy_logging_server.ts#L112-L116

Reply:
also we can ignore the problem and fix it when move logging infra to NP

Reply:
TBH, I'm leaning towards punting on this for now as well. It's not a new issue and we need to do the work to complete the migration to the new logger anyways, where we can fix this there.